ra/ra.go

```diff
@@ -1366,19 +1366,32 @@ func (ra *RegistrationAuthorityImpl) checkCertificatesPerNameLimit(ctx context.C
 }
 
 func (ra *RegistrationAuthorityImpl) checkCertificatesPerFQDNSetLimit(ctx context.Context, names []string, limit ratelimit.RateLimitPolicy, regID int64) error {
-	count, err := ra.SA.CountFQDNSets(ctx, &sapb.CountFQDNSetsRequest{
+	issuanceTimestampsInWindow, err := ra.SA.FQDNSetTimestampsForWindow(ctx, &sapb.CountFQDNSetsRequest{
 		Domains: names,
 		Window:  limit.Window.Duration.Nanoseconds(),
 	})
 	if err != nil {
 		return fmt.Errorf("checking duplicate certificate limit for %q: %s", names, err)
 	}
+
 	names = core.UniqueLowerNames(names)
 	threshold := limit.GetThreshold(strings.Join(names, ","), regID)
-	if count.Count >= threshold {
+	if threshold <= 0 {
+		// No limit configured.
+		return nil
+	}
```

Comment:
1) We can do this GetThreshold call before the SA query on line 1369 to save some effort; and
2) With this in place, the `if issuanceCount <= 0` clause on line 1385 becomes unnecessary

Reply:
> We can do this GetThreshold call before the SA query on line 1369 to save some effort

Ah, yeah I meant to do this on my last commit and it must have slipped my mind.